CONTAINER_INTERFACE: clarify that /proc/sys can be writable with namespacing

bluca opened the pull request:
When user and network namespaces are enabled, the kernel
makes the global keys read-only, and makes the namespaced
ones available for the guest already.

poettering commented:
uts namespaces are relevant for /proc/sys/kernel/hostname. And the pid namespace stuff for the pid props. I don't know the details. I'd probably use some vague wording here and not get too much into detail, given this also changed in the kernel in the past. I mean, userns and netns deserve explicit mention, but the other ones are not that clear.

bluca commented:
> uts namespaces are relevant for /proc/sys/kernel/hostname. And the pid namespace stuff for the pid props. I don't know the details. I'd probably use some vague wording here and not get too much into detail, given this also changed in the kernel in the past. I mean, userns and netns deserve explicit mention, but the other ones are not that clear.

Added a mention of uts/pid too - the feedback from implementors was that this part is not very clear. So we should either be specific, or drop mentions of /proc/sys, otherwise it's just confusing for people writing container managers.

poettering commented on docs/CONTAINER_INTERFACE.md:
```diff
@@ -22,10 +22,12 @@ manager, please consider supporting the following interfaces.
+   to prevent the container from altering the host kernel's configuration
+   settings. (As a special exception, if your container has network namespaces
+   enabled, feel free to make `/proc/sys/net/` writable. If it also has user,
+   uts and pid namespaces enabled, the entire `/proc/sys` can be left writable).
```
ipc too

bluca replied:
thanks, updated
